### PR TITLE
US 5 // Filter resources that have path

### PR DESCRIPTION
**Pull request by Dinika**
Screenshot from 2023-07-04 14-24-42

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added necessary unit and integration tests.
- [X] I have added screenshots (if applicable), in the comment section.

### CODE REVIEW

File: src/subapps/dataExplorer/PredicateSelector.tsx [hunk 1]
```diff
@@ -142,11 +176,47 @@ const getPathsForResource = (
...
+
+  return criteria === 'exists' ? false : true;
+};
+
 export const isPathMissing = (
   resource: { [key: string]: any },
   path: string
 ): boolean => {
-  if (path in resource) {
+  if (path in resource && path !== '') {
```
Comment (bilalesi) [inline src/subapps/dataExplorer/PredicateSelector.tsx:219]: Is there situation where the path is empty string ?
If so is it better to filter paths when you generate them (just a suggestion)

File: src/subapps/dataExplorer/PredicateSelector.tsx [hunk 2]
```diff
@@ -91,9 +119,15 @@ export const PredicateSelector: React.FC<Props> = ({
...
+export const DOES_NOT_EXIST = 'Does not exist';
+export const EXISTS = 'Exists';
 export const CONTAINS = 'Contains';
-export type PredicateFilterT = typeof EMPTY_VALUE | typeof CONTAINS | null;
+
+export type PredicateFilterT =
+  | typeof DOES_NOT_EXIST
+  | typeof EXISTS
+  | typeof CONTAINS
+  | null;
```
Comment (bilalesi) [inline src/subapps/dataExplorer/PredicateSelector.tsx:130]: I suggest use this to be more `Typescripty`
`const PREDICATE_VALUES = [DEFAULT_OPTION,  DOES_NOT_EXIST, EXISTS, CONTAINS] as const`;
`type TPredicateValues = typeof PREDICATE_VALUES;`
and then 
`type TPredicateValue = TPredicateValues[0 | 1 | 2] or TPredicateValues[number]`
and then select by index when you need it `TPredicateValues[index]`

File: src/subapps/dataExplorer/DataExplorer.tsx [hunk 3]
```diff
@@ -4,41 +4,23 @@ import { notification } from 'antd';
...
+import { PredicateSelector } from './PredicateSelector';
 
 export interface DataExplorerConfiguration {
   pageSize: number;
   offset: number;
   orgAndProject?: [string, string];
-  predicatePath: string | null;
-  predicateFilter: PredicateFilterT | null;
-  predicateValue: string | null;
+  predicateFilter: ((resource: Resource) => boolean) | null;
```
Comment (bilalesi) [inline src/subapps/dataExplorer/DataExplorer.tsx:16]: I think it make sense if the `predicateFilter` can be rename it to `predicate`

**Review by bilalesi — APPROVED** [review]

File: src/subapps/dataExplorer/PredicateSelector.tsx (hunk 2 above)
Comment (Dinika) [reply src/subapps/dataExplorer/PredicateSelector.tsx:130]: bon idee! I'll do that in a new PR.

File: src/subapps/dataExplorer/DataExplorer.tsx (hunk 3 above)
Comment (Dinika) [reply src/subapps/dataExplorer/DataExplorer.tsx:16]: this too.

**Pull request merged**